staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go

```diff
@@ -905,17 +905,14 @@ func TestApplyPruneObjectsWithAllowlist(t *testing.T) {
 				"namespace/test-apply pruned",
```

Comment:
This test case should be updated as well.

Reply:
I think it should prune the non-namespaced items when a namespace and a allow list is not specified.

It's interesting that this test case is failing, but this test under the [prune_test.go](https://github.com/kubernetes/kubernetes/blob/cd04f47e8b287068e8bf0bfae6596b2606d54929/staging/src/k8s.io/kubectl/pkg/util/prune/prune_test.go#L58C1-L66C4) is passing.

```
{
	mapper:             &testRESTMapper{},
	pr:                 []Resource{},
	namespaceSpecified: false,
	expectedns:         14,
	expectednns:        2,
	expectederr:        nil,
},
```

Reply:
> It's interesting that this test case is failing, but this test under the [prune_test.go](https://github.com/kubernetes/kubernetes/blob/cd04f47e8b287068e8bf0bfae6596b2606d54929/staging/src/k8s.io/kubectl/pkg/util/prune/prune_test.go#L58C1-L66C4) is passing.

This case should pass. 

- If the namespace is not specified, default 14 namespace resources and 2 none-namespaced resources will be pruned.
- If the namespace is specified, only the default 14 namespace resources will be pruned.

Reply:
> > It's interesting that this test case is failing, but this test under the [prune_test.go](https://github.com/kubernetes/kubernetes/blob/cd04f47e8b287068e8bf0bfae6596b2606d54929/staging/src/k8s.io/kubectl/pkg/util/prune/prune_test.go#L58C1-L66C4) is passing.
> 
> This case should pass.
> 
> * If the namespace is not specified, default 14 namespace resources and 2 none-namespaced resources will be pruned.
> * If the namespace is specified, only the default 14 namespace resources will be pruned.

Yes that's true, but this test under the [apply_test.go](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/119687/pull-kubernetes-unit/1697893585954803712) is failing.

```
=== FAIL: vendor/k8s.io/kubectl/pkg/cmd/apply TestApplyPruneObjectsWithAllowlist/prune_without_namespace_and_allowlist_should_delete_resources_that_are_not_in_the_specified_file (0.03s)
    apply_test.go:1040: expected output to contain "namespace/test-apply pruned", but it did not. Actual Output:
        replicationcontroller/test-rc unchanged
        configmap/test-cm pruned
        replicationcontroller/test-rc2 pruned
```

It should have pruned the namesapce.

Reply:
Sorry, after investigation, I think we have made a mistake here.

https://github.com/kubernetes/kubernetes/blob/3c4a813d9c475909b50d1ea510ae5f34b5c4b847/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go#L462-L463

- the namespace will default to `default` if not set.

https://github.com/kubernetes/kubernetes/blob/3c4a813d9c475909b50d1ea510ae5f34b5c4b847/staging/src/k8s.io/kubectl/pkg/cmd/apply/prune.go#L74-L75

namespaceSpecified `o.Namespace != ""` will always be true. 

The warning message will be there for everyone(no matter users specified namespace for the `kubectl apply --prune`) if I understand correctly.

Reply:
/hold

Reply:
I wonder what should we do now?

Reply:
@pacoxu I was referring to the original issue and I ended up in this PR. where does the validation for `the namespace will default to default if not set.` happens. I dont see it in DefaultNamespace() IIUC. Could you please provide the pointer to that